packages/models/src/resource-binding/manager.ts

```diff
@@ -28,43 +30,65 @@ export class Manager extends Actor<object, MessagePayload<ResourceBindingManager
   }
 
   private update(pollingInterval = 1000) {
+    let semaphore = 0
     return setInterval(async () => {
-      if (this.#tipBlockNumber.gt(0) && this.#tipBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
-        this.updateStore(await this._dataSource.getBlock(this.#tipBlockNumber.toHexString()))
+      semaphore++
+      if (semaphore == 1) {
+        if (this.#buffer.hasReadyStore()) {
+          this.updateBuffer()
+        }
+        if (this.#tipBlockNumber.gt(0) && this.#tipBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
+          this.updateStore(await this._dataSource.getBlock(this.#tipBlockNumber.toHexString()))
+        }
       }
+
+      semaphore--
     }, pollingInterval)
   }
 
-  private updateStore(block: Block) {
-    if (!this.#lastBlock || BI.from(block.header.number).gt(BI.from(this.#lastBlock.header.number))) {
-      const changes = this.filterCells(block)
+  private updateBuffer() {
+    const stores = this.#buffer.popAll()
+    for (const changes of stores) {
       for (const [store, input, data] of changes) {
-        const upgrade: { witness: HexString; cell: Cell }[] = []
-        for (const [cell, witness] of data) {
-          upgrade.push({ witness, cell })
-          if (cell.outPoint) {
-            this.#registryOutPoint.set(outPointToOutPointString(cell.outPoint), store)
-          }
-        }
+        this.updateCellChanges(store, input, data)
+      }
+    }
+  }
 
-        if (upgrade.length > 0) {
-          this.sendMessage(store, 'update_cells', upgrade)
-        }
+  private updateCellChanges(store: ResourceBindingRegistry, input: Input[], data: CellChangeData[]) {
+    const upgrade: UpdateStorageValue[] = []
+    for (const [cell, witness] of data) {
+      upgrade.push({ witness, cell })
+      if (cell.outPoint) {
+        this.#registryOutPoint.set(outPointToOutPointString(cell.outPoint), store)
+      }
+    }
 
-        if (input.length > 0) {
-          this.sendMessage(
-            store,
-            'remove_cell',
-            input.map((v) => outPointToOutPointString(v.previousOutput)),
-          )
-        }
+    if (upgrade.length > 0) {
+      this.sendMessage(store, 'update_cells', upgrade)
+    }
+
+    if (input.length > 0) {
+      this.sendMessage(
+        store,
+        'remove_cell',
+        input.map((v) => outPointToOutPointString(v.previousOutput)),
+      )
+    }
+  }
+
+  private updateStore(block: Block) {
+    if (!this.#lastBlock || BI.from(block.header.number).gt(BI.from(this.#lastBlock.header.number))) {
```

Comment:
```suggestion
    if (BI.from(block.header.number).gte(BI.from(this.#lastBlock?.header.number ?? 0))) {
```

Reply:
I think `if (!this.#lastBlock || BI.from(block.header.number).gt(BI.from(this.#lastBlock.header.number))) {` is more readable than `if (BI.from(block.header.number).gte(BI.from(this.#lastBlock?.header.number ?? 0))) {`.